src/dialog/savedqueries/dlgsavedquerieseditor.cpp

```diff
@@ -0,0 +1,50 @@
+#include <QSqlTableModel>
+
+#include "dialog/savedqueries/dlgsavedquerieseditor.h"
+#include "dialog/savedqueries/savedqueriestablemodel.h"
+#include "library/libraryfeature.h"
+#include "library/trackcollection.h"
+
+DlgSavedQueriesEditor::DlgSavedQueriesEditor(LibraryFeature* pFeature,
+                                             TrackCollection* pTrackCollection,
+                                             QWidget* parent)
+        : QDialog(parent),
+          m_pTrackCollection(pTrackCollection),
+          m_pFeature(pFeature) {
+    setupUi(this);
+    SavedQueriesTableModel *pSaveModel = 
+            new SavedQueriesTableModel(m_pFeature, 
+                                       m_pTrackCollection->getSavedQueriesDAO(),
+                                       parent);
+    tableView->setModel(pSaveModel);
+    for (int i = 0; i < SavedQueryColumns::NUM_COLUMNS; ++i) {
+        tableView->setColumnHidden(i, pSaveModel->isColumnInternal(i));
+    }
+    
+    tableView->setSelectionBehavior(QAbstractItemView::SelectRows);
+    tableView->setSelectionMode(QAbstractItemView::SingleSelection);
+    tableView->verticalHeader()->hide();
+    
+    connect(pushRemove, SIGNAL(pressed()), this, SLOT(removeQuery()));
+}
+
+void DlgSavedQueriesEditor::accept() {
+    tableView->model()->submit();
+    QDialog::accept();
+}
+
+void DlgSavedQueriesEditor::removeQuery() {
+    QItemSelectionModel* model = tableView->selectionModel();
```

Comment:
The link to the Google style guide is broken there. It seems the only exception to AAA is "don't use auto in function declarations", which I agree with, and is not a c++11 feature anyway... Other uses of the explicit type should be the exception and not the rule. In this case auto is clearly preferred, right?